shared/appengine.go

```diff
@@ -24,6 +24,31 @@ import (
 	"google.golang.org/appengine/user"
 )
 
+type clientsImpl struct {
+	cloudtasks *cloudtasks.Client
+}
+
+// Clients is a singleton containing heavyweight (e.g. with connection pools)
+// clients that should be bound to the runtime instead of each request in order
+// to be reused. They are initialized and authenticated at startup using the
+// background context; each request should use its own context.
+var Clients clientsImpl
+
+// Init initializes all clients in Clients. If an error is encountered, it
+// returns immediately without trying to initialize the remaining clients.
+func (c *clientsImpl) Init(ctx context.Context) (err error) {
+	c.cloudtasks, err = cloudtasks.NewClient(ctx)
+	return err
+}
+
+// Close closes all clients in Clients. It must be called once and only once
+// before the server exits. Do not use any AppEngine APIs afterwards.
```

Comment:
You mean Google Cloud client library ?

Reply:
Hmm I was trying to refer to any API in this file, just to be safe (though some APIs don't use these clients for now). Let me rephrase it

```suggestion
// before the server exits. Do not use AppEngineAPI afterwards.
```